cypress/tests/ImageElement.spec.ts

```diff
@@ -68,10 +68,60 @@ describe("ImageElement", () => {
           .should("have.text", "deco");
       });
 
-      it(`caption – should map decorations passed from the parent editor correctly when they move`, () => {
+      it(`caption – should map decorations passed from the parent editor correctly when they move – add text before`, () => {
         addImageElement();
-        const text = `caption deco{leftarrow}{leftarrow}{leftarrow}{leftarrow}{leftarrow} more text`;
-        typeIntoElementField("caption", text);
+        typeIntoElementField(
+          "caption",
+          "start deco{leftarrow}{leftarrow}{leftarrow}{leftarrow}{leftarrow} more"
+        );
+        getElementRichTextField("caption")
+          .find(".TestDecoration")
+          .should("have.text", "deco");
+      });
+
+      it(`caption – should map decorations passed from the parent editor correctly when they move – add text after`, () => {
+        addImageElement();
+        typeIntoElementField("caption", "caption deco more text");
+        getElementRichTextField("caption")
+          .find(".TestDecoration")
+          .should("have.text", "deco");
+      });
+
+      it(`caption – should map decorations passed from the parent editor correctly when they move – remove text before`, () => {
+        addImageElement();
+        typeIntoElementField(
+          "caption",
+          "start deco{leftarrow}{leftarrow}{leftarrow}{leftarrow}{leftarrow}{backspace}{backspace}"
+        );
+        getElementRichTextField("caption")
+          .find(".TestDecoration")
+          .should("have.text", "deco");
+      });
+
+      it(`caption – should map decorations passed from the parent editor correctly when they move – remove text after`, () => {
+        addImageElement();
+        typeIntoElementField(
+          "caption",
+          "caption deco more text {backspace}{backspace}"
+        );
+        getElementRichTextField("caption")
+          .find(".TestDecoration")
+          .should("have.text", "deco");
+      });
+
+      it(`caption – should map decorations passed from the parent editor correctly when they move – add text in following field`, () => {
+        addImageElement();
+        typeIntoElementField("caption", "caption deco");
+        typeIntoElementField("altText", "text");
+        getElementRichTextField("caption")
+          .find(".TestDecoration")
+          .should("have.text", "deco");
+      });
+
+      it(`caption – should map decorations passed from the parent editor correctly when they move – remove text in following field`, () => {
```

Comment:
Is it worth adding test cases where we might have a decoration with a range which might change? We could then add characters within that range and outside of it, to ensure it handles those cases correctly.

Reply:
Can't hurt! Followed up in #145.